src/doc/src/reference/semver.md

```diff
@@ -1164,6 +1165,61 @@ Mitigation strategies:
 * Document the platforms and environments you specifically support.
 * Test your code on a wide range of environments in CI.
 
+<a id="new-lints"></a>
+### Minor: introducing new lints
+
+Some changes to a library may cause new lints to be triggered in users of that library.
+This should generally be considered a compatible change.
+
+```rust,ignore,dont-deny
+// MINOR CHANGE
+
+///////////////////////////////////////////////////////////
+// Before
+pub fn foo() {}
+
+///////////////////////////////////////////////////////////
+// After
+#[deprecated]
+pub fn foo() {}
+
+///////////////////////////////////////////////////////////
+// Example use of the library that will safely work.
+
+fn main() {
+    updated_crate::foo(); // Warning: use of deprecated function
+}
+```
+
+Beware that it may be possible for this to technically cause a project to fail if they have explicitly denied the warning, and the updated crate is a direct dependency.
+Denying warnings should be done with care and the understanding that new lints may be introduced over time.
+However, library authors should be cautious about introducing new warnings and may want to consider the potential impact on their users.
+
+The following lints are examples of those that may be introduced when updating a dependency:
+
+* [`deprecated`][deprecated-lint] — Introduced when a dependency adds the [`#[deprecated]` attribute][deprecated] to an item you are using.
+* [`unused_must_use`] — Introduced when a dependency adds the [`#[must_use]` attribute][must-use-attr] to an item where you are not consuming the result.
+* [`unused_unsafe`] — Introduced when a dependency *removes* the `unsafe` qualifier from a function, and that is the only unsafe function called in an unsafe block.
+
+Additionally, updating `rustc` to a new version may introduce new lints.
+
+Transitive dependencies which introduce new lints should not usually cause a failure because Cargo uses [`--cap-lints`](../../rustc/lints/levels.html#capping-lints) to suppress all lints in dependencies.
+
+Mitigating strategies:
+* Options for dealing with denying warnings:
+    * Understand you may need to deal with resolving new warnings whenever you update your dependencies.
+    * Place `deny(warnings)` behind a [feature][Cargo features], for example `#![cfg_attr(feature = "deny-warnings", deny(warnings))]`.
```

Comment:
Do we recommend `#![deny(warnings)]` elsewhere?  I've generally viewed it as an anto-pattern.

Problems with `#![[deny(warnings)]`
- Makes you dependent on the version of Rust used
- Blocks testing refactors for resolving warnings

Even cargo's approach of making it part of the test compilation has been a point of frustration at times.

As for the workarounds, a deny-warnigns feature flag means that it is now part of your API compatibility (and yes, I've seen a related feature used and a caller activate it for no reason).

Instead, if the user is doing `-Dwarnings` in CI, a workaround would be to instead do `-D warnings --allow deprecated`

Reply:
No I don't think it is recommended anywhere. I think those who choose to deny warnings usually understand the downsides of it. 

And yea, using a feature is suboptimal. Using `-D` flags is also suboptimal since it breaks the cache. 

I added some text here to note the flags option and some of the downsides.

Having better lint controls would improve these options quite a bit (I would remove both these suggestions if it was available).

Reply:
> Using -D flags is also suboptimal since it breaks the cache.

At least how I run it, this doesn't matter.  For local development, you don't use `-D`, only for a dedicated CI job.